src/dataset/nuscenes_dataset.py

```diff
@@ -1,26 +1,173 @@
+        scene_names = nuscenes.utils.splits.create_splits_scenes()[f'mini_{subset}' if mini_version else subset]
         self.scenes = [scene for scene in self.nusc.scene if scene['name'] in scene_names]
-        self.img_ids = [sample_data_token
-                        for scene in self.scenes for sample_data_token in self._get_sample_data_list(scene)]
+        self.videos = self._get_videos(self.scenes, self.sensors)
+        self.video_indexer = dataset_utils.VideosIndexer(self.videos)
+
+    def _pre_process_input(self, sample_data_token):
+        frame_sample_data = self.nusc.get('sample_data', sample_data_token)
+        cal_sensor_data, pose_data, trans_matrix = self._compute_calibration_data(frame_sample_data)
+        _, boxes, camera_intrinsic = self.nusc.get_sample_data(sample_data_token,
+                                                               box_vis_level=geometry_utils.BoxVisibility.ANY)
+        calibration = self._get_calibration_matrix(camera_intrinsic)
+
+        image_info = {'file_name':         frame_sample_data['filename'],
+                      'calib':             calibration.tolist(),
+                      'trans_matrix':      trans_matrix.tolist(),
+                      'width':             frame_sample_data['width'],
+                      'height':            frame_sample_data['height'],
+                      'pose_record_trans': pose_data['translation'],
+                      'pose_record_rot':   pose_data['rotation'],
+                      'cs_record_trans':   cal_sensor_data['translation'],
+                      'cs_record_rot':     cal_sensor_data['rotation']}
+
+        annotations = self._get_annotations(boxes, calibration, camera_intrinsic, trans_matrix)
+
+        frame_path = os.path.join(self.dataset_root, frame_sample_data['filename'])
+        read_frame = cv2.imread(frame_path)
+
+        return read_frame, annotations, image_info
+
+    def _compute_calibration_data(self, frame_sample_data):
+        cal_sensor_data = self.nusc.get('calibrated_sensor', frame_sample_data['calibrated_sensor_token'])
+        pose_data = self.nusc.get('ego_pose', frame_sample_data['ego_pose_token'])
+        car_to_global = geometry_utils.transform_matrix(pose_data['translation'], Quaternion(pose_data['rotation']),
+                                                        inverse=False)
+        sensor_to_car = geometry_utils.transform_matrix(cal_sensor_data['translation'],
+                                                        Quaternion(cal_sensor_data['rotation']),
+                                                        inverse=False)
+        trans_matrix = np.dot(car_to_global, sensor_to_car)
+        return cal_sensor_data, pose_data, trans_matrix
+
+    def _get_calibration_matrix(self, camera_intrinsic):
+        calibration = np.eye(4, dtype=np.float32)
+        calibration[:3, :3] = camera_intrinsic
+        calibration = calibration[:3]
```

Comment:
What is the dimension of `camera_intrinsic`? I think you could simply do `return camera_intrinsic[:3]`

Reply:
camera_intrinsic is (3,3). The result must be (3,4). Basically, we need to add an extra 0 column to camera_intrinsic.

Reply:
Ah, sorry, I see it now. Thanks for the clarification.